public/js/utils/pretty-print.js

```diff
@@ -0,0 +1,37 @@
+const { getValue } = require("../feature");
+const { workerTask } = require("./utils");
+const { isJavaScript } = require("./source");
+const assert = require("./assert");
+
+let prettyPrintWorker;
+prettyPrintWorker = new Worker(
+  getValue("baseWorkerURL") + "pretty-print-worker.js"
+);
+
+function destroyWorker() {
+  prettyPrintWorker.terminate();
+  prettyPrintWorker = null;
+}
+
+const _prettyPrint = workerTask(prettyPrintWorker, "prettyPrint");
+
+async function prettyPrint({ source, sourceText, url }) {
+  const contentType = sourceText ? sourceText.contentType : null;
+  const indent = 2;
+
+  assert(
+    isJavaScript(source.url, contentType),
+    "Can't prettify non-javascript files."
+  );
+
+  return await _prettyPrint({
+    url,
+    indent,
+    source: sourceText.text
+  });
+}
+
+module.exports = {
+  prettyPrint,
+  destroyWorker
+};
```

Comment:
i like that this module cleans up the actions, as well as is consistent with source map